code/modules/clothing/suits/marine_armor.dm

```diff
@@ -288,51 +288,265 @@ var/list/squad_colors = list(rgb(230,25,25), rgb(255,195,45), rgb(200,100,200),
+		handle_chem_cooldown(B18_BRUTE_CODE)
+		dose_administered = TRUE
+
+	if(wearer.getOxyLoss() > B18_automed_damage && !B18_oxy_cooldown)
+		var/dexalinplus = REAGENTS_OVERDOSE * 0.5 - (wearer.reagents.get_reagent_amount("dexalinplus") + 0.5)
+		var/inaprovaline = REAGENTS_OVERDOSE * 2 - (wearer.reagents.get_reagent_amount("inaprovaline") + 0.5)
+		if(dexalinplus < REAGENTS_OVERDOSE*0.5)
+			wearer.reagents.add_reagent("dexalinplus",dexalinplus)
+		if(inaprovaline < REAGENTS_OVERDOSE)
+			wearer.reagents.add_reagent("inaprovaline",inaprovaline)
+		if(tricordrazine < REAGENTS_OVERDOSE)
+			wearer.reagents.add_reagent("tricordrazine",tricordrazine)
+		details +=("Low blood oxygen detected. Reoxygenating preparation administered. <b>Dosage: Dexalin Plus: [dexalinplus]U | Inaprovaline: [inaprovaline]U | Tricordrazine: [tricordrazine]U</br>")
+		B18_oxy_cooldown = world.time + B18_CHEM_COOLDOWN
+		handle_chem_cooldown(B18_OXY_CODE)
+		dose_administered = TRUE
+
+	if(wearer.getToxLoss() > B18_automed_damage && !B18_tox_cooldown)
+		var/dylovene = REAGENTS_OVERDOSE - (wearer.reagents.get_reagent_amount("dylovene") + 0.5)
+		var/spaceacillin = REAGENTS_OVERDOSE - (wearer.reagents.get_reagent_amount("spaceacillin") + 0.5)
+		if(dylovene < REAGENTS_OVERDOSE)
+			wearer.reagents.add_reagent("dylovene",dylovene)
+		if(spaceacillin < REAGENTS_OVERDOSE)
+			wearer.reagents.add_reagent("spaceacillin",spaceacillin)
+		if(tricordrazine < REAGENTS_OVERDOSE)
+			wearer.reagents.add_reagent("tricordrazine",tricordrazine)
+		details +=("Significant blood toxicity detected. Chelating agents and curatives administered. <b>Dosage: Dylovene: [dylovene]U | Spaceacillin: [spaceacillin]U | Tricordrazine: [tricordrazine]U</b></br>")
+		B18_tox_cooldown = world.time + B18_CHEM_COOLDOWN
+		handle_chem_cooldown(B18_TOX_CODE)
+		dose_administered = TRUE
+
+	if(wearer.traumatic_shock > B18_automed_pain && !B18_pain_cooldown)
+		var/oxycodone = REAGENTS_OVERDOSE * 0.66 - (wearer.reagents.get_reagent_amount("oxycodone") + 0.5)
+		var/tramadol = REAGENTS_OVERDOSE - (wearer.reagents.get_reagent_amount("tramadol") + 0.5)
+		if(oxycodone < REAGENTS_OVERDOSE)
+			wearer.reagents.add_reagent("oxycodone",oxycodone)
+		if(tramadol < REAGENTS_OVERDOSE)
+			wearer.reagents.add_reagent("tramadol",tramadol)
+		details +=("User pain at performance impeding levels. Painkillers administered. <b>Dosage: Oxycodone: [oxycodone]U | Tramadol: [tramadol]U</b></br>")
+		B18_pain_cooldown = world.time + B18_CHEM_COOLDOWN
+		handle_chem_cooldown(B18_PAIN_CODE)
+		dose_administered = TRUE
+
+	if(dose_administered)
+		playsound(src,'sound/items/hypospray.ogg', 25, 0, 1)
+		details +=("Estimated [B18_CHEM_COOLDOWN/600] minute replenishment time for each dosage.")
+		to_chat(wearer, "<span class='notice'>\icon [src] beeps:</br> [details.Join(" ")]</span>")
+
+/obj/item/clothing/suit/storage/marine/specialist/proc/handle_chem_cooldown(code = B18_BRUTE_CODE)
```

Comment:
add a silent var here too, specially if no cure was administered during the process.

Reply:
There is no cooldown without administration.

Reply:
I mean, in case the damage was outputted but no cure was given (read: the wearer's reagents associated to the chemical exceed the safe parameters).